bd-client-stats/src/stats.rs

```diff
@@ -191,25 +200,33 @@ impl<F: SerializedFileSystem> Uploader<F> {
 
     // If there is a pending upload, first attempt to re-upload.
     let pending_upload = if self.fs.exists(&*PENDING_STATS_UPLOAD_FILE).await {
-      if let Ok(pending_request) = self
+      match self
         .fs
         .read_compressed_protobuf_file::<StatsUploadRequest>(&*PENDING_STATS_UPLOAD_FILE)
         .await
       {
-        Some(pending_request)
-      } else {
-        // We failed to read the data, so the file must be bad. This could happen if we change
-        // the schema in an incompatible way or if the file is corrupt. Delete the file and
-        // accept the loss of this upload.
-
-        // TODO(snowp): Technically we could end up in a situation here where we are stuck
-        // trying to delete a bad file which then spams the error handler.
-        // TODO(snowp): Track how often we delete the file here.
-        handle_unexpected(
-          self.fs.delete_file(&*PENDING_STATS_UPLOAD_FILE).await,
-          "delete pending stats upload",
-        );
-        None
+        Ok(pending_request) => {
+          log::debug!(
+            "found pending stats upload, attempting to re-upload: {}",
+            pending_request
+          );
+          Some(pending_request)
+        },
+        Err(e) => {
+          // We failed to read the data, so the file must be bad. This could happen if we change
+          // the schema in an incompatible way or if the file is corrupt. Delete the file and
+          // accept the loss of this upload.
+
+          // TODO(snowp): Technically we could end up in a situation here where we are stuck
+          // trying to delete a bad file which then spams the error handler.
```

Comment:
We can remove this at this point I think since we limit how many times we report an error deeper in the error handler